prometheusbp/promtest/spec.go

```diff
@@ -0,0 +1,97 @@
+package promtest
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+
+	"github.com/reddit/baseplate.go/errorsbp"
+)
+
+var (
+	errPrefix         = errors.New("the prefix is not at the beginning of the metric name")
+	errLength         = errors.New("metric name should have a minimum of 3 parts, like prefix_name_suffix")
+	errCount          = errors.New("wrong metric count for prefix")
+	errPrometheusLint = errors.New("problem with Prometheus GatherAndLint")
+)
+
+// ValidateSpec validates that the Prometheus metrics being exposed conform to
+// the baseplate spec. This test will validate all metrics that begin with the
+// metricPrefix and will validate the number of metrics with that prefix match
+// wantMetricCount.
+func ValidateSpec(t *testing.T, metricPrefix string, wantMetricCount int) {
+	var batchErrs errorsbp.Batch
+	gotMetricCount, errs := validateSpec(metricPrefix)
+	if len(errs) > 0 {
+		batchErrs.Add(errs...)
+	}
```

Comment:
> If an error is also a Batch, its underlying error(s) will be added instead of the Batch itself.

so `validateSpec` can return a `batch.Compile()` as a single error and you can `batchErrs.AddPrefix("validate spec", err)` here